build2/KoreBuild.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+echo -e "\033[1;32mPreparing KoreBuild 2.0...\033[0m"
+
+DOTNET_INSTALL="$DIR/../build/dotnet/dotnet-install.sh"
+DOTNET_VERSION_DIR="$DIR/../build"
+
+KOREBUILD_LOG="$PWD/korebuild.log"
+[ ! -e "$KOREBUILD_LOG" ] || rm "$KOREBUILD_LOG"
+
+ensure_dotnet() {
+    if test `uname` = Darwin; then
+        versionFileName="cli.version.darwin"
+    else
+        versionFileName="cli.version.unix"
+    fi
+    versionFile="$DOTNET_VERSION_DIR/$versionFileName"
+    version=$(<$versionFile)
+
+    [ -z "$KOREBUILD_DOTNET_CHANNEL" ] && KOREBUILD_DOTNET_CHANNEL=preview
+    [ -z "$KOREBUILD_DOTNET_VERSION" ] && KOREBUILD_DOTNET_VERSION=$version
+
+    if [ ! -z "$KOREBUILD_SKIP_RUNTIME_INSTALL" ]; then
+        echo "Skipping runtime installation because KOREBUILD_SKIP_RUNTIME_INSTALL is set"
+
+        # Add .NET installation directory to the path if it isn't yet included.
+        # Add to the _end_ in case preferred .NET CLI is not in the default location.
+        [[ ":$PATH:" != *":$DOTNET_INSTALL_DIR:"* ]] && export PATH="$PATH:$DOTNET_INSTALL_DIR"
+    else
+        # Need to set this variable because by default the install script
+        # requires sudo
+        export DOTNET_INSTALL_DIR=~/.dotnet
+        export KOREBUILD_FOLDER="$(dirname $DIR)"
```

Comment:
Interestingly, the original `KoreBuild.sh` script sets it twice and to [different](https://github.com/aspnet/KoreBuild/blob/d7473ac46c470112fbd9e4d5fe3fe93f2d4305d9/build/KoreBuild.sh#L54) [values](https://github.com/aspnet/KoreBuild/blob/d7473ac46c470112fbd9e4d5fe3fe93f2d4305d9/build/KoreBuild.sh#L95) so any build that does depend on it is going to be a bit wonky.